File: articles/data-measurement-reproducibility-life-sciences/python/run_all.py
"""
Run all Python data, measurement, and reproducibility workflows.

Run:
    python python/run_all.py
"""

from __future__ import annotations

import subprocess
import sys
from pathlib import Path


SCRIPT_DIR = Path(__file__).resolve().parent

SCRIPTS = [
    "reproducibility_core.py",
    "measurement_quality_summary.py",
    "schema_validation.py",
    "uncertainty_budget.py",
    "provenance_manifest.py",
    "checksum_manifest.py",
    "qc_flag_summary.py",
    "reproducibility_report.py",
    "workflow_audit.py",
]


def main() -> None:
    for script in SCRIPTS:
        print(f"\n=== Running {script} ===")
        subprocess.run([sys.executable, str(SCRIPT_DIR / script)], check=True)


if __name__ == "__main__":
    main()

File: articles/r-biological-data-analysis-visualization/r/run_all.R
# Run all R workflows.

scripts <- c(
  "r/01_measurement_quality_summary.R",
  "r/02_assay_visualization.R",
  "r/03_ecological_diversity.R",
  "r/04_dose_response_visualization.R",
  "r/05_reproducibility_manifest.R"
)

for (script in scripts) {
  message("\n=== Running ", script, " ===")
  source(script, local = new.env())
}
